packages/runtime/container-runtime/src/connectionTelemetry.ts

```diff
@@ -63,10 +69,35 @@ class OpPerfTelemetry {
         this.deltaManager.on("disconnect", () => {
             this.opSendTimeForLatencyStatisticsForMsnStatistics = undefined;
             this.clientSequenceNumberForLatencyStatistics = undefined;
+            this.opTimeSittingInInboundQueue = undefined;
+            this.durationSittingInInboundQueue = undefined;
             this.connectionOpSeqNumber = undefined;
             this.firstConnection = false;
         });
 
+        this.deltaManager.outbound.on("push", (messages) => {
+            for (const msg of messages) {
+                if (msg.type === MessageType.Operation &&
+                    this.clientSequenceNumberForLatencyStatistics === msg.clientSequenceNumber) {
+                    assert(this.opTimeSittingInInboundQueue === undefined,
+                        "OpTimeSittingInboundQueue should be undefined");
+                    assert(this.durationSittingInInboundQueue === undefined,
+                        "durationSittingInInboundQueue should be undefined");
+                     this.opTimeSittingInInboundQueue = Date.now();
+                }
+            }
+        });
+
+        this.deltaManager.inbound.on("push", (message: ISequencedDocumentMessage) => {
+            if (this.clientId === message.clientId &&
+                message.type === MessageType.Operation &&
+                this.clientSequenceNumberForLatencyStatistics === message.clientSequenceNumber) {
+                assert(this.opTimeSittingInInboundQueue !== undefined, "opTimeSittingInInboundQueue should be defined");
```

Comment:
Ends up Vlad was right, this assert will be hit in some scenarios where batching is turned off. At least until the loader changes are in. 

@vladsud - I'm inclined to move the assert condition back to the if (this.opTimeSittingInInboundQueue !== undefined ), as it was before,  until the loader is in.  Then we could probably move back to the assert.

Thoughts ?